packages/spear-cli/src/file/InMemoryFileManipulator.ts

```diff
@@ -0,0 +1,116 @@
+import { FileManipulatorInterface, InMemoryFile } from "../interfaces/FileManipulatorInterface";
+import { SiteMapURL } from "../interfaces/MagicInterfaces";
+import { DefaultSettings } from "../interfaces/SettingsInterfaces";
+
+export class InMemoryFileManipulator implements FileManipulatorInterface {
+    files: InMemoryFile[]
+    settingsObject: DefaultSettings
+    constructor(files: InMemoryFile[], settingsObject) {
+        this.files = files
+    }
+
+    // In memory environment, we don't care the file encoding.
+    // These files is based on UTF-8.
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    readFileSync(path: string, encode: string): string {
+        for (const file of this.files) {
+            if (file.path === path) return file.content
+        }
+        return ""
+    }
+
+    readFileSyncAsBuffer(path: string): Buffer {
+        for (const file of this.files) {
+            if (file.path === path) return Buffer.from(file.content, 'base64')
+        }
+        return Buffer.from("", 'base64')
+    }
+
+    readdirSync(path: string): string[] {
+        const ret = [] as string[]
+        this.files.forEach(file => {
+            if (file.path.includes(path)) ret.push(file.path.replace(path, ""))
+        })
+        return ret
+    }
+
+    existsSync(path: string): boolean {
+        for (const file of this.files) {
+            if (file.path === path) return true
+        }
+        return false
+    }
+
+    isDirectory(path: string): boolean {
+        for (const file of this.files) {
+            if (file.path === path && file.content === "") return true
```

Comment:
It is possible to have a file with empty string in it, so I think it would be like this. 🤔 

```suggestion
            if (file.path === path && file.content === null) return true
```

Reply:
Ooops! Your suggestion is right!